snuba/query/mql/parser.py

```diff
@@ -49,18 +49,27 @@
 
 # The parser returns a bunch of different types, so create a single aggregate type to
 # capture everything.
-MQLSTUFF = Dict[str, Union[str, List[SelectedExpression], List[Expression]]]
+MQLSTUFF = Dict[str, Union[str, list[SelectedExpression], list[Expression]]]
 logger = logging.getLogger("snuba.mql.parser")
 
 
 @dataclass
 class InitialParseResult:
     expression: SelectedExpression | None = None
-    groupby: List[SelectedExpression] | None = None
-    conditions: List[Expression] | None = None
+    formula: str | None = None
+    parameters: list[InitialParseResult] | None = None
```

Comment:
If parameters are present, is it guaranteed that formula also exists? If so, do you think it makes sense to codify that? E.g.
```
@dataclass
class InitialParseResult:
    expression: SelectedExpression | None = None
    formula: ParseFormula | None = None

@dataclass
class ParseFormula
    name: str
    parameters: list[InitialParseResult]
```

Reply:
I don't think it's guaranteed, we could have a Formula with no parameters. I think keeping things in a single structure keeps things a little simpler for now. Particularly the typing.